File: dev/20190415/k2works/fizzbuzz/src/main/java/org/hiroshima_arc/domain/type/FizzBuzzType01.java
package org.hiroshima_arc.domain.type;

public class FizzBuzzType01 implements IFizzBuzzType {
    @Override
    public String generate(int number) {
        boolean fizz = isFizz(number);
        boolean buzz = isBuzz(number);
        if (fizz && buzz) return "FizzBuzz";
        if (fizz) return "Fizz";
        if (buzz) return "Buzz";
        return Integer.toString(number);
    }

    private boolean isBuzz(int number) {
        return number % 5 == 0;
    }

    private boolean isFizz(int number) {
        return number % 3 == 0;
    }
}

File: dev/20190902/k2works/app/FizzBuzz/Domain/FizzBuzzType01.cs
namespace FizzBuzz.Domain
{
    public class FizzBuzzType01: FizzBuzzType 
    {
        public override string Generate(int number)
        {
            var isFizz = number % 3 == 0;
            var isBuzz = number % 5 == 0;

            if (isFizz && isBuzz)
            {
                return FizzBuzz.FIZZ_BUZZ;
            }

            if (isFizz)
            {
                return FizzBuzz.FIZZ;
            }

            if (isBuzz)
            {
                return FizzBuzz.BUZZ;
            }

            return number.ToString();
        }
    }
}
